lib/spack/spack/detection/path.py

```diff
@@ -100,18 +107,17 @@ def libraries_in_ld_and_system_library_path(
 
 
 def libraries_in_windows_paths(path_hints: List[str]) -> Dict[str, str]:
-    path_hints.extend(spack.util.environment.get_path("PATH"))
-    search_paths = llnl.util.filesystem.search_paths_for_libraries(*path_hints)
+    search_hints = path_hints or spack.util.environment.get_path("PATH")
```

Comment:
This function uses defaults when `path_hints=[]`, which runs counter to the behavior of other functions we've discussed in this PR, as well as the parents - is that intended?

I think docstrings would help clarify the intent.